[IDP-42] DDS: Forcepoint Security Service Edge: Crawler Integration v1.0.0 (#19360)
DataDog/integrations-core: forcepoint_security_service_edge/README.md

## Overview

[Forcepoint Security Service Edge][1] simplifies security at the edge by delivering safe access and data protection. Security Service Edge (SSE) eliminates gaps in coverage by unifying policy configuration, enforcement and reporting under a single platform.

This integration ingests the following logs:

- **Cloud Logs (CloudSummary, CloudAudit)**: It contains logs related to scanning results of each file.
- **Access Logs**: It contains logs related to various application activities.
- **Admin Logs**: It contains all admin events performed within the admin portal.
- **Health Logs (HealthProxy, HealthApi, HealthSystem)**: It contains logs related to system, api and proxy health. 

> 

Author: Have updated as per suggestion

Forcepoint Security Service Edge integration gathers these logs and forwards them to Datadog for seamless analysis. Datadog leverages its built-in log pipelines to parse and enrich these logs, facilitating easy search and detailed insights. With preconfigured out-of-the-box dashboards, the integration offers clear visibility into activities within the Forcepoint Security Service Edge platform. Additionally, it includes ready-to-use Cloud SIEM detection rules for enhanced monitoring and security.

## Setup

### Generate OAuth Token in Forcepoint Security Service Edge:
1. Login to the Forcepoint ONE Security Service Edge Platform.
2. Navigate to **SETTINGS > API Interface > OAuth**.
3. **REST API OAuth Configuration** page opens which allows you to add and configure different levels of API permissions.

> 4. To add a new configuration, click the **green** plus icons.

Author: Have updated as per suggestion

> 5. On the **Edit Application** dialog, fill out the information as mentioned below:

Author: Have updated as per suggestion

    a. **Name**: Name for the new application configuration

    b. **Permissions**: Select **Access your Forcepoint logs (logs api)** option.

>     c. **Permitted User Group**: Select as per your requirement. Default is **All**.

Author: Have updated as per suggestion

>     d. Click **Ok** to save the changes.
>     - You will now see your application added to the list, but still listed as **Pending** under status.

Reviewer: Double checking, in the UI is it Ok or OK?

```suggestion
    d. Click **Ok** to save the changes. You should see your application added to the list, but listed as **Pending** under status.
```

Author: In the UI it's `Ok`. Have updated statement as per suggestion

> 6. Select the name of your application in the **Application** column to  go into the **Edit Application**.

Reviewer: ```suggestion
6. Select the name of your application in the **Application** column to go into the **Edit Application**.
```

Author: Have updated as per suggestion

>     a. On the **Edit Application** dialog, you will need the **Token Authorization URL** to authorize your current permission and get the access token.

Reviewer: ```suggestion
    a. On the **Edit Application** dialog, click the **Token Authorization URL** to authorize your current permission and get the access token.
```

Author: Have updated as per suggestion

>     b. Click on the URL and it will take you to the **Requested Access** page allowing you to **Approve** or **Deny** the application permission settings. Again you will need to send this URL to each permitted user and have them **Approve** their access.

Reviewer: Is "this URL" the Token Authorization URL?

```suggestion
    b. On the **Requested Access** page send this URL to each permitted user and have them **Approve** their access. The **Requested Access** page allows you to **Approve** or **Deny** the application permission settings.
```

Author: Have updated as per suggestion

> 7. After you approve, you will be given an **Access Token** that is unique to that user  and that the user must keep. This access token will be required to configure integration in datadog. The token is valid forever and must be included in each request for authorization.

Reviewer: ```suggestion
7. After the user approves, they are given an **Access Token** that is unique to that user. The user must keep this access token, it is required to configure integrations in Datadog. The token is valid forever and must be included in each request for authorization.
```

8. Once access has been approved, you will notice that **Status** is changed to **Authorized**.

> For reference: [Setting up an OAuth token Documentation][2]

Reviewer: ```suggestion
For more information, see the [Setting up an OAuth token][2] documentation.
```

Author: Have updated as per suggestion

### Connect your Forcepoint Security Service Edge Account to Datadog

1. Add your Access Token.
   | Parameters          | Description                                                                           |
   | ------------------- | ------------------------------------------------------------------------------------- |
   | Access Token       | Access token from Forcepoint Security Service Edge                         |

> 2. Click the Save button to save your settings.

Reviewer: ```suggestion
2. Click **Save**.
```

Author: Have updated as per suggestion

## Overview

[Forcepoint Security Service Edge][1] simplifies security at the edge by delivering safe access and data protection. Security Service Edge (SSE) eliminates gaps in coverage by unifying policy configuration, enforcement and reporting under a single platform.

This integration ingests the following logs:

> - **Cloud Logs (CloudSummary, CloudAudit)**: Logs related to scanning results of each file.

Reviewer 2: Cloud Logs are "logs related to scanning results of each file"? That doesn't make sense. What files? Please update the text here to better reflect what is collected with these logs.

Author: We have updated description to `Logs related to the current status of files in cloud applications and scan results for each file in the account`.
